packages/ui/src/components/Visualization/Custom/ContextMenu/ItemAddStep.tsx

```diff
@@ -29,7 +46,14 @@ export const ItemAddStep: FunctionComponent<ItemAddStepProps> = (props) => {
 
     /** Update entity */
     entitiesContext.updateEntitiesFromCamelResource();
-  }, [catalogModalContext, entitiesContext, props.mode, props.vizNode]);
+
+    setTimeout(() => {
+      const result = props.vizNode.data.path
+        ? findVisualizationNode(controller.getGraph(), props.vizNode.data.path)
+        : null;
+      controller.fireEvent(SELECTION_EVENT, [result?.getNextNode()?.id]);
```

Comment:
There are a few cases in which `getNextNode` doesn't yield the desired node:
1. When adding a step in `from`
2. Preprending a step from any given node

I'll try to check if we can get the new node id directly.

Reply:
Thanks, I will wait for your changes in https://github.com/lordrip/kaoto/commit/3d668cd273d2e2de8dfa5e0526d946c213b9246e before adding more capabilities